client/client.go

```diff
@@ -24,112 +19,68 @@ func main() {
 		os.Exit(1)
 	}
 
-	args := os.Args
-	var certFile string
-	var keyFile string
-	var host string
-	var endpoint string
-
-	body, err := ReadArgs(args, &endpoint, &certFile, &keyFile, &host)
-	if err != nil {
-		fmt.Println(err.Error())
-		os.Exit(1)
-	}
-
-	var request *http.Request
-
-	switch endpoint {
+	startCommand := flag.NewFlagSet("start", flag.ExitOnError)
+	startCertFile := startCommand.String("cert", "cert.pem", "path to cert file")
+	startCaCertFile := startCommand.String("cacert", "", "path to cacert file")
+	startKeyFile := startCommand.String("key", "key.pem", "path to key file")
+	startHost := startCommand.String("host", "localhost", "endpoint of request")
+	startExec := startCommand.String("command", "", "command to exec")
+
+	stopCommand := flag.NewFlagSet("stop", flag.ExitOnError)
+	stopCertFile := stopCommand.String("cert", "cert.pem", "path to cert file")
+	stopCaCertFile := stopCommand.String("cacert", "", "path to cacert file")
+	stopKeyFile := stopCommand.String("key", "key.pem", "path to key file")
+	stopHost := stopCommand.String("host", "localhost", "endpoint of request")
+	stopTaskID := stopCommand.String("task_id", "", "task_id of process")
+
+	statusCommand := flag.NewFlagSet("status", flag.ExitOnError)
+	statusCertFile := statusCommand.String("cert", "cert.pem", "path to cert file")
+	statusCaCertFile := statusCommand.String("cacert", "", "path to cacert file")
+	statusKeyFile := statusCommand.String("key", "key.pem", "path to key file")
+	statusHost := statusCommand.String("host", "localhost", "endpoint of request")
+	statusTaskID := statusCommand.String("task_id", "", "task_id of process")
+
+	switch os.Args[1] {
 	case "start":
-		request, err = StartRequest(body, host)
+		startCommand.Parse(os.Args[2:])
 	case "stop":
-		request, err = StopRequest(body, host)
+		stopCommand.Parse(os.Args[2:])
 	case "status":
-		request, err = StatusRequest(body, host)
+		statusCommand.Parse(os.Args[2:])
 	default:
-		err = errors.New("invalid command")
+		fmt.Println("invalid command")
 		os.Exit(1)
 	}
 
-	client, err = newClient(certFile, keyFile, certFile)
-	if err != nil {
-		fmt.Println(err.Error())
+	var request *http.Request
+	var permission Permission
+
+	var err error
+
+	if startCommand.Parsed() {
+		permission = Permission{*startCertFile, *startKeyFile, *startCaCertFile}
```

Comment:
nit: Given that `Permission` only has `string` attributes, it is relatively easy to set the wrong attributes using only the value list syntax. You should strive to use keyed initialization syntax instead:
```go
permission = Permission{
	CertFile: *startCertFile,
	KeyFile: *startKeyFile,
	CaCertFile: *startCaCertFile,
}
```